packages/hdwallet-metamask-multichain/src/adapter.ts

```diff
@@ -0,0 +1,60 @@
+import * as core from "@shapeshiftoss/hdwallet-core";
+import { enableShapeShiftSnap, shapeShiftSnapInstalled } from "@shapeshiftoss/metamask-snaps-adapter";
+
+import { SNAP_ID } from "./common";
+import { mipdstore } from "./mipdStore";
+import { MetaMaskMultiChainHDWallet } from "./shapeshift-multichain";
+
+export class MetaMaskAdapter {
+  keyring: core.Keyring;
+  providerRdns: string;
+
+  private constructor(keyring: core.Keyring, providerRdns: string) {
+    this.keyring = keyring;
+    this.providerRdns = providerRdns;
+  }
+
+  public static useKeyring(keyring: core.Keyring, providerRdns: string) {
+    return new MetaMaskAdapter(keyring, providerRdns);
+  }
+
+  public async initialize(): Promise<number> {
+    return Object.keys(this.keyring.wallets).length;
+  }
+
+  public async pairDevice(): Promise<MetaMaskMultiChainHDWallet | undefined> {
+    const maybeEip6963Provider = mipdstore.findProvider({ rdns: this.providerRdns });
+    if (!maybeEip6963Provider) throw new Error("EIP-6963 provider not found");
```

Comment:
we lost the onboarding logic here, do we want to trigger mm onboarding for `io.metamask` so things behave the same?

Reply:
Waiting on a reply from product here, intent is to use wc instead since deep linking is broken half/most of the time depending on platforms

Reply:
Disregard me - confirmed we want to keep the current behaviour *fully* as in, always display providers even if not installed (in the same order as currently), only appending the ones we don't currently have already